File: java_programs/DynamicAllocation.java
import java.io.FileWriter;
import java.io.IOException;

public class DynamicAllocation {
    public static void main(String[] args) throws IOException {
        int NUM_ELEMENTS = 1000000;
        long totalTime = 0;

        for (int test = 0; test < 10; test++) {
            long start = System.nanoTime();
            Integer[] array = new Integer[NUM_ELEMENTS];
            for (int i = 0; i < NUM_ELEMENTS; i++) {
                array[i] = i;
            }
            long end = System.nanoTime();
            totalTime += (end - start);
        }

        FileWriter writer = new FileWriter("./results/java_dynamic_allocation_results.txt");
        writer.write(String.format("Average: %.6f ns\n", totalTime / 10.0));
        writer.close();
    }
}

File: python_programs/dynamic_allocation.py
import time

NUM_ELEMENTS = 1000000
total_time = 0

for _ in range(10):
    start = time.perf_counter()
    array = [None] * NUM_ELEMENTS
    for i in range(NUM_ELEMENTS):
        array[i] = i
    end = time.perf_counter()
    total_time += (end - start) * 1e9

with open("./results/python_dynamic_allocation_results.txt", "w") as f:
    f.write(f"Average: {total_time / 10:.6f} ns\n")
